Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp

```diff
@@ -117,10 +117,20 @@ void DNSResolveQueueCFNet::performDNSLookup(const String& hostname, Ref<Completi
     nw_parameters_set_context(parameters.get(), context.get());
     RetainPtr resolver = adoptCF(nw_resolver_create_with_endpoint(hostEndpoint.get(), parameters.get()));
 
-    nw_resolver_set_update_handler(resolver.get(), dispatch_get_main_queue(), makeBlockPtr([resolver, completionHandler = WTFMove(completionHandler)] (nw_resolver_status_t status, nw_array_t resolvedEndpoints) mutable {
+    auto timer = makeUnique<Timer>([resolver, completionHandler]() {
+        WTF::ensureOnMainThread([resolver, completionHandler]() {
+            nw_resolver_cancel(resolver.get());
+            completionHandler->complete(makeUnexpected(DNSError::Cancelled));
+        });
+    });
+    timer->startOneShot(Seconds(60));
```

Comment:
```suggestion
    timer->startOneShot(60_s);
```
But usually we would introduce a `static constexpr Seconds timeoutForDNSResolution = 60_s;`